### PR TITLE
[bitnami/redis] Tests for Standalone, Sentinel and Replica topologies 

### PR DESCRIPTION
**Pull request by superaleks**
Signed-off-by: Aleksandra Lukic <alukic@vmware.com>

**Description of the change**

Created infrastructure tests for three different Redis topologies, written using Goss.

**Benefits**

We will be able to test the Redis Helm chart whenever there is a change. 

**Possible drawbacks**

Tests will have to be maintained and improved with time. 

**Applicable issues**

  - fixes #

**Additional information**

A new workflow is added for this pipeline. The workflow contains 4 pipelines: validation pipeline with Helm Lint and Trivy actions, which is a precondition for the individual topologies. 

**Checklist**

- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by beltran-rubo** [comment]
The changes LGTM, please @mpermar confirm if the files structure is ok for now.

File: .github/workflows/vib-redis.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+name: 'vib'
+on: # rebuild any PRs and main branch changes
+  pull_request_target:
+    branches:
+      - master
+      - bitnami:master
```
Comment (juan131) [inline .github/workflows/vib-redis.yaml:6]: Should't it be enough with 'master'?

File: .github/workflows/vib-redis.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{github.event.pull_request.head.ref}}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+      - uses: vmware-samples/vmware-image-builder-examples/vib-action@main
+        with: 
+          pipeline: redis/replicas/vib-platform-verify.json
+        env: 
+          TARGET_PLATFORM: 7ddab896-2e4e-4d58-a501-f79897eba3a0           
+
```
Comment (juan131) [inline .github/workflows/vib-redis.yaml:75]: Please address the 'no newline at end of file' warning. Recommendation: configure your IDE to do it for you:
- https://stackoverflow.com/questions/44704968/visual-studio-code-insert-new-line-at-the-end-of-files

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for this great PR! Please check my comments when you have a chance

File: .github/workflows/vib-redis.yaml (hunk 1 above)
Comment (beltran-rubo) [reply .github/workflows/vib-redis.yaml:6]: Yes right, not needed. In any case we will work on a full review of the GitHub action & folders structure.

File: .github/workflows/vib-redis.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+on: # rebuild any PRs and main branch changes
+  pull_request_target:
+    branches:
+      - master
+      - bitnami:master
+    types:
+      - assigned 
+      - opened 
+      - synchronize 
+      - reopened
```
[the remark this answers is not in the document]
Comment (beltran-rubo) [reply .github/workflows/vib-redis.yaml:11]: It is pending to be reviewed with the team for the best workflow.

File: .vib/redis/replicas/vib-platform-verify.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          }
+        ]
+      },
+      "verify": {
+        "context": {
+          "resources": {
+            "url": "{SHA_ARCHIVE}",
+            "path": "/bitnami/redis"
+          },
+          "runtime_parameters": "YXV0aDoKICBwYXNzd29yZDogInRlc3RfcGFzc3dvcmQiCm1hc3RlcjoKICBjb250YWluZXJTZWN1cml0eUNvbnRleHQ6CiAgICBlbmFibGVkOiB0cnVlCiAgICBydW5Bc1VzZXI6IDEwMDIgCiAgcGVyc2lzdGVuY2U6CiAgICBwYXRoOiAvZGF0YS1wZXJzaXN0ZW5jZQpyZXBsaWNhOgogIHJlcGxpY2FDb3VudDogMgogIGNvbnRhaW5lclNlY3VyaXR5Q29udGV4dDoKICAgIGVuYWJsZWQ6IHRydWUKICAgIHJ1bkFzVXNlcjogMTAwMg==",
```
[the remark this answers is not in the document]
Comment (beltran-rubo) [reply .vib/redis/replicas/vib-platform-verify.json:22]: We can clarify in a readme for contributors for now.

A possible approach is to use something like jsonnet base64("values"), but it also has the con of pre-processing the files before sending the API query.

File: .github/workflows/vib-redis.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mpermar) [reply .github/workflows/vib-redis.yaml:11]: Yeah, those types come from me and a demo. There is no particular reason. There are actually tons more: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

Perhaps opened/reopened are just enough. Synchronize I'm not even sure what is.

[the remark this answers is not in the document]
Comment (superaleks) [reply .github/workflows/vib-redis.yaml]: Changed to have only opened/reopened, as suggested.

File: .github/workflows/vib-redis.yaml [hunk 5]
```diff
@@ -0,0 +1,71 @@
+name: 'vib'
+on: # rebuild any PRs and main branch changes
+  pull_request_target:
+    branches:
+      - master
+    types: 
+      - opened 
+      - reopened
```
Comment (juan131) [inline .github/workflows/vib-redis.yaml:8]: We should include "synchronize" that means "run the action if a new commit arrives":
- ref: https://frontside.com/blog/2020-05-26-github-actions-pull_request/

**Review by juan131 — APPROVED** [review]
LGTM! Thanks @superaleks

**Pull request merged**